app/models/zone.rb

```diff
@@ -197,8 +197,7 @@ def ems_clouds
 
   # @return [Array<ExtManagementSystem>] All emses that can collect Capacity and Utilization metrics
   def ems_metrics_collectable
-    supported_ems_types = ExtManagementSystem.subclasses_supports?(:metrics).map(&:name)
-    ext_management_systems.where(:type => supported_ems_types).select { |e| e.supports?(:metrics) }
+    ext_management_systems.supporting(:metrics)
```

Comment:
https://github.com/ManageIQ/manageiq/pull/21553

Reply:
Nice!

Reply:
We discussed this one on slack, but we still need to evaluate the instance-level supports blocks (e.g. azure class supports metrics but instances can not support it due to insights service not being enabled)

Reply:
I wonder if we should formalize the "class-level" supports and the "instance-level" supports.  I'm a little concerned that we're changing the default behavior here from instance-level to class-level and we might end up introducing bugs like this almost was.  cc @Fryguy

Reply:
Well, it feel to me like we have a kind of validation embedded in this supports concept.
Does this ems type support this concept (class level) and does this ems in particular support this feature (instance level)
It gets confusing since we only sometimes implement instance level checking.

Reply:
Yeah we have to be very careful about changing something from:
```ruby
Class.all.select { |obj| obj.supports?(:feature) }
```
and
```ruby
Class.descendants.select { |klass| klass.supports?(:feature) }
```

One only checks the class supports it, the other checks that the instance supports it and they aren't interchangeable